molvis/gaussian/python/processors/cubesourcemolvis.py

```diff
@@ -0,0 +1,155 @@
+ # DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ # ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ # (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ # LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ # ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #
+ #################################################################################
+ 
+
+import inviwopy as ivw
+import ivwdataframe as df
+import ivwmolvis
+import gaussianutil
+
+from pathlib import Path
+
+class CubeSourceMolvis(ivw.Processor):
+    def __init__(self, id, name):
+        ivw.Processor.__init__(self, id, name)
+
+        self.volumeOutport = ivw.data.VolumeOutport("chargedensity")
+        self.addOutport(self.volumeOutport, owner=False)
+
+        self.meshOutport = ivw.data.MeshOutport("atoms")
+        self.addOutport(self.meshOutport)
+
+        self.dataframeOutport = df.DataFrameOutport("atomInformation")
+        self.addOutport(self.dataframeOutport)
+
+        self.addOutport(ivwmolvis.MolecularStructureOutport("molecule"))
+
+        self.cubeFilePath = ivw.properties.FileProperty("cube", "cube")
+        self.addProperty(self.cubeFilePath)
+
+        self.dataRange = ivw.properties.DoubleMinMaxProperty(
+            "dataRange", "Data Range", 0.0, 1.0, -1.70e308, 1.79e308)
+        self.addProperty(self.dataRange)
+        self.dataRange.semantics = ivw.properties.PropertySemantics("Text")
+        self.dataRange.readOnly = True
+
+        self.useCustomRange = ivw.properties.BoolProperty(
+            "useCustomRange", "Overwrite Data Range", False)
+        self.addProperty(self.useCustomRange)
+
+        self.customDataRange = ivw.properties.DoubleMinMaxProperty(
+            "customDataRange", "Custom Data Range", 0.0, 1.0, -1.70e308, 1.79e308)
+        self.addProperty(self.customDataRange)
+        self.properties.customDataRange.semantics = ivw.properties.PropertySemantics(
+            "Text")
+
+        self.flipSign = ivw.properties.BoolProperty(
+            "flipSign", "Flip Sign of Charge", False)
+        self.addProperty(self.flipSign)
+
+        self.centerData = ivw.properties.BoolProperty(
+            "centerData", "Center Data", True)
+        self.addProperty(self.centerData)
+
+        self.pm = ivw.PickingMapper(self, 1, lambda x: self.callback(x))
+
+    @staticmethod
+    def processorInfo():
+        return ivw.ProcessorInfo(
+    		classIdentifier = "org.inviwo.CubeSourceMolvis", 
+    		displayName = "Cube Source Molvis",
+    		category = "Python",
+    		codeState = ivw.CodeState.Stable,
+    		tags=ivw.Tags([ivw.Tag.PY, ivw.Tag("Cube"), 
+                           ivw.Tag("Gaussian"), ivw.Tag("Volume"), ivw.Tag("Mesh")])
```

Comment:
Is cube a file format?
Should there also be a molecule, or similar, tag?

Reply:
yes, it is a file format containing volumetric data and atomic positions (http://www.paulbourke.net/dataformats/cube/).
Could potentially add `MolVis` as a tag (https://github.com/inviwo/modules/blob/9bf8d3375d8d5885869a7858870a88455e274a3c/molvis/molvispython/python/processors/MolecularStructureSource.py#L66)